demisto_sdk/commands/format/update_generic.py

```diff
@@ -220,95 +221,76 @@ def regex_matching_key(self, field, schema_keys):
-                    LooseVersion(NEW_FILE_DEFAULT_5_5_0_FROMVERSION) and file_type != PLAYBOOK:
-                if self.assume_yes:
-                    self.data[self.from_version_key] = NEW_FILE_DEFAULT_5_5_0_FROMVERSION
-                else:
-                    set_from_version = str(
-                        input(f"\nYour current fromversion is: '{self.data.get(self.from_version_key)}'. Do you want "
-                              f"to set it to '5.5.0'? Y/N ")).lower()
-                    if set_from_version in ['y', 'yes']:
-                        self.data[self.from_version_key] = NEW_FILE_DEFAULT_5_5_0_FROMVERSION
-
-        # If there is an existing file in content repo
-        else:
-            # If current file does not have fromversion key
-            if self.from_version_key not in self.data:
-
-                # If user entered specific from version key to be set
-                if from_version:
-                    self.data[self.from_version_key] = from_version
-
-                # If existing file already have a fromversion key, copy its value to current file
-                elif self.from_version_key in self.old_file:
-                    self.data[self.from_version_key] = self.old_file[self.from_version_key]
-
-                # Otherwise add fromversion key to current file and set to default 1.0.0
-                else:
-                    self.data[self.from_version_key] = OLD_FILE_DEFAULT_1_FROMVERSION
+        version_to_set = None
+        current_fromversion_value = self.data.get(self.from_version_key)
+        if self.verbose:
+            click.echo('Setting fromVersion field')
+
+        # If user asked specific fromVersion to set.
+        if self.from_version:
+            version_to_set = self.from_version
+        # If there is already a fromVersion in file and its 5.5.0 and it is new supported integration (issue #35130).
+        elif current_fromversion_value and current_fromversion_value == VERSION_5_5_0 and self.is_new_supported_integration(
+                file_type):
+            version_to_set = current_fromversion_value
+        # If there is fromVersion in the old_file(repo).
+        elif self.old_file.get(self.from_version_key) and not current_fromversion_value:
+            version_to_set = self.old_file.get(self.from_version_key)
+        # If there isn't current fromVersion, ask the user to generate it.
+        elif not current_fromversion_value:
+            # If it is old content type the supported version is 5.5.0.
+            version_to_set = VERSION_5_5_0 if self.old_content_type(file_type) else self.set_default_from_version(
+                default_from_version)
+        # If the content type default fromVersion is higher than the current fromVersion.
+        elif default_from_version and server_version_compare(current_fromversion_value, default_from_version) < 0:
+            if self.assume_yes or self.ask_user():
+                version_to_set = default_from_version
```

Comment:
continuing this conversation: https://github.com/demisto/demisto-sdk/pull/1934#discussion_r826063357
Please elaborate, why can't we unify them? They look the same to me (different conditions but same result). in particular, you don't handle here the case where `default_from_version` is lower than `GENERAL_DEFAULT_FROMVERSION` (we should set the fromversion to `GENERAL_DEFAULT_FROMVERSION` in this case) - which is already handled in `set_default_from_version()`.